### DISCUSSION TITLE
[BUG] Set-PnPGroupPermissions stopped working today!!!

### DISCUSSION THREAD

**Discussion by mvanders** [pnp/powershell#3490] [General]
The following code never had any issue to run since years...

Set-PnPGroupPermissions -Identity $MembersGroupName -List "SiteAssets" -AddRole @('Read') # FOR NEW

As of today, the code doesn't work anymore :-(
THIS IS FOR US VERY URGENT!

**What is the version of the Cmdlet module you are running? 1.12.0**
**Which operating system/environment are you running PnP PowerShell on?**
[X ] Windows
 Linux
 MacOS
 Azure Cloud Shell
 Azure Functions
 Other : please specify

THIS ISSUE WAS ALREADY POSTEN LAST YEAR TOO! 
https://github.com/pnp/powershell/issues/2244

**Comment by veronicageek** [comment]
You may want to upgrade to the latest `2.2.0` for the major version (or even a nightly one), because I have no issue on my side.

_NOTE: You will need PS 7.2 at the minimum for PnP PowerShell v2_

![image](https://github.com/pnp/powershell/assets/25181757/44d2da7c-4b3a-4b14-9396-65d8f27b6204)

**Comment by mvanders** [comment]
For several reasons I need to stick to PS5.1. Is there than no solution?

Sent from Outlook for iOS<https://aka.ms/o0ukef>
________________________________

**Comment by veronicageek** [comment]
You may try to use CSOM directly, but we do not support PS5 anymore.

**Comment by mvanders** [comment]
Could you provide with the csom code?

Sent from Outlook for iOS<https://aka.ms/o0ukef>
________________________________

**Reply by gautamdsheth** [reply]
What is the error that you are facing ? 

Have you tried minimal code like the one mentioned above ?

**Reply by mvanders** [reply]
I believe this is minimal code, not?

Set-PnPGroupPermissions -Identity $MembersGroupName -List "SiteAssets" -AddRole @('Read')

Results in:
The error:
Set-PnPGroupPermissions : Method not found: 'System.Runtime.CompilerServices.ConfiguredValueTaskAwaitable`1
Enumerator.MoveNextAsync()'.
At line:372 char:9
    Set-PnPGroupPermissions -Identity $DeputiesGroupName -List "S ...
    ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
CategoryInfo : InvalidOperation: (:) [Set-PnPGroupPermissions], PSInvalidOperationException
FullyQualifiedErrorId : InvalidOperation,PnP.PowerShell.Commands.Principals.SetGroupPermissions

**Reply by mvanders** [reply]
Can nobody help me???

**Reply by CallumCrowley** [reply]
Are you using other PowerShell libraries at the same time as PnP.PowerShell? Sometimes, I have experienced problems where multiple PowerShell libraries being used in the same session have conflicting dependencies (libraries) that they rely upon.

**Reply by AndersRask** [reply]
Later builds have fixed a lot of the issues with dll collisions since it uses ALC bridging pattern.

https://github.com/pnp/powershell/issues/3207

**Reply by CallumCrowley** [reply]
@AndersRask @mvanders is using 1.12, which was last updated November 2022. The fix for #3207 is from August 2023.

**Reply by AndersRask** [reply]
Yes, another reason to upgrade then :)

**Reply by mvanders** [reply]
No, I am not using other libraries, but indeed on PS5.1 (for several reasons).
So the only way to make that command work again is upgrade to PS 7.X? 
Not very fair in my opinion, because I'm sure if I'll do that, I will run into other issues I gues...

**Reply by AndersRask** [reply]
Could you provide us with a full stack trace? I just tested the function on 5.1 and 1.12 and it works fine.

`$error[0] | fl * -force`

Also try and enable trace and attach a logfile

`Set-PnPTraceLog -On -LogFile traceoutput.txt`

**Comment by mvanders** [comment]
@AndersRask Upon your request, I tried again today...
 
PS C:\GitHub\ACEA\PowerShell\Scripts> Set-PnPGroupPermissions -Identity $MembersGroupName -List "Documents" -AddRole @('OpenWithCreateAlerts')

Gives an error:
Set-PnPGroupPermissions : Method not found: 'System.Runtime.CompilerServices.ConfiguredValueTaskAwaitable`1<Boolean> 
Enumerator.MoveNextAsync()'.
At line:1 char:1
+ Set-PnPGroupPermissions -Identity $MembersGroupName -List "Documents" ...
+ ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
    + CategoryInfo          : InvalidOperation: (:) [Set-PnPGroupPermissions], PSInvalidOperationException
    + FullyQualifiedErrorId : InvalidOperation,PnP.PowerShell.Commands.Principals.SetGroupPermissions

The full stack trace ($error[0] | fl * -force)
PSMessageDetails      :
Exception             : System.Management.Automation.PSInvalidOperationException: Method not found: 
                        'System.Runtime.CompilerServices.ConfiguredValueTaskAwaitable`1<Boolean> Enumerator.MoveNextAsync()'.
                           at PnP.PowerShell.Commands.Base.PnPConnectedCmdlet.ProcessRecord()
                           at System.Management.Automation.CommandProcessor.ProcessRecord()
TargetObject          : 
CategoryInfo          : InvalidOperation: (:) [Set-PnPGroupPermissions], PSInvalidOperationException
FullyQualifiedErrorId : InvalidOperation,PnP.PowerShell.Commands.Principals.SetGroupPermissions
ErrorDetails          : 
InvocationInfo        : System.Management.Automation.InvocationInfo
ScriptStackTrace      : at <ScriptBlock>, <No file>: line 1
PipelineIterationInfo : {}

**Comment by mvanders** [comment]
IMPORTANT !!!

• After searching and a lot of trial and error, I found this ...
	[Add-PnPPage : Method not found: 'System.Runtime.CompilerServices.ConfiguredValueTaskAwaitable`1<Boolean> Enumerator.MoveNextAsync()' · Issue #120026 · microsoft/vscode (github.com)](https://github.com/microsoft/vscode/issues/120026)

**Saying it works in ISE for 5.1 but NOT in VS code and apparently upgrading to 7.X doesn't help!**
So I start re-using ISE... thanks Microsoft :-S

**Reply by CallumCrowley** [reply]
That's a few years old, I've used VS Code with PnP.PowerShell and PowerShell 7.2 without any problem. The new [VS Code PowerShell](https://marketplace.visualstudio.com/items?itemName=ms-vscode.PowerShell) helps a lot.
